cmd/process-exporter/main.go

```diff
@@ -355,7 +357,7 @@ func main() {
 		matchnamer = namemapper
 	}
 
-	pc, err := NewProcessCollector(*procfsPath, *children, *threads, matchnamer, *recheck, *debug)
+	pc, err := NewProcessCollector(*procfsPath, *children, *threads, *smaps, matchnamer, *recheck, *debug)
```

Comment:
I think we've reached or exceeded my threshold for number of constructor parameters.  I'm going to switch this to use an options struct after your work is merged.  Or you could do so if you're inclined, either way.